experiments/snippets/skuidCustom/skuidCustom.js

```diff
@@ -0,0 +1,1344 @@
+Date.getDateFromSFDate = function (dateStr) {
+	if (dateStr === undefined || dateStr === null || typeof dateStr !== 'string') {
+		return undefined;
+	}
+	
+	const dateArr = dateStr.split('-');
+	
+	if (dateArr.length !== 3) {
+		return undefined;
+	}
+	
+	const year = dateArr[0];
+	const month = dateArr[1];
+	const day = dateArr[2];
+	
+	const yearInt = Number(year);
+	const monthInt = Number(month);
+	const dayInt = Number(day);
+	
+	const d = new Date(yearInt, monthInt - 1, dayInt);
+	
+	return d;
+};
+
+// Date.isLeapYear(year)
+// Checks if a year is a leap year, returns true or false
+Date.isLeapYear = function (year) {
+	return (year % 100 === 0) ? (year % 400 === 0) : (year % 4 === 0);
+};
+Date.prototype.isLeapYear = function () { 
+	return Date.isLeapYear(this.getFullYear()); 
+};
+
+// Date.getDaysInMonth(year,month)
+// Given a year and a month returns the number of days in that month. Accounts for leap years.
+Date.getDaysInMonth = function (year, month) {
+	return [31, (Date.isLeapYear(year) ? 29 : 28), 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][month];
+};
+Date.prototype.getDaysInMonth = function () { 
+	return Date.getDaysInMonth(this.getFullYear(), this.getMonth());
+};
+
+// Date.addMonths(months)
+// Adds a number of months to a date, returns the updated date
+Date.prototype.addMonths = function (months) {
+	const n = this.getDate();
+	this.setDate(1);
+	this.setMonth(this.getMonth() + months);
+	this.setDate(Math.min(n, this.getDaysInMonth()));
+	return this;
+};
+
+// Date.addDays(days)
+// adds a number of days to a date, returns the updated date
+Date.prototype.addDays = function (days) {
+	const date = new Date(this.valueOf());
+	date.setDate(date.getDate() + days);
+	return date;
+};
+
+// Object.byString(ourObject,fieldToGet);
+// Able to get object's sub-objects with a single string variable
+// instead of needing dot notation between multiple separate variables
+// eg: Object.byString(ourObject,'Field__r.SubField__c')
+// instead of needing to do ourObject['Field__r']['SubField__c'] or ourObject.Field__r.SubField__c
+Object.byString = function (ourObject, fieldToGet) {
+	let s = fieldToGet;
+	let o = ourObject;
+	s = s.replace(/\[(\w+)\]/g, '.$1'); // convert indexes to properties
+	s = s.replace(/^\./, ''); // strip a leading dot
+	const a = s.split('.');
+	for (let i = 0, n = a.length; i < n; ++i) {
+		const k = a[i];
+		if (k in o) {
+			o = o[k];
+		} else {
+			return;
+		}
+	}
+	return o;
+};
+
+// skuid.custom.fixCurrency(number)
+// remove anything past 2 digits after the decimal point for a number
+// necessary to do after doing any mathematical operations between 2 floating point numbers
+skuid.custom.fixCurrency = function (num) {
+	// return Math.floor(num*100)/100;
+	return parseFloat(num.toFixed(2));
+};
+
+// skuid.custom.clearArray(array)
+// force-clears the data out of an array which may still otherwise have data in it
+skuid.custom.clearArray = function (array) {
+	while (array.length) {

```

Comment:
the fastest, most performant way to do this is just: `array.length = 0`

Reply:
Removed from skuidCustom